pilot/pkg/status/state.go

```diff
@@ -249,7 +255,7 @@ func ReconcileStatuses(current map[string]interface{}, desired Progress, generat
 	if err != nil {
 		// the status field is in an unexpected state.
 		scope.Warn("Encountered unexpected status content.  Overwriting status.")
-		scope.Debugf("Encountered unexpected status content.  Overwriting status: %v", current["status"])
+		scope.Debugf("Encountered unexpected status content.  Overwriting status: %v", current.Status)
```

Comment:
nit: prefer `if scope.DebugEnable { debug } else {warn}`. otherwise you get 2x logs